Source/WebCore/platform/graphics/Damage.h

```diff
@@ -171,75 +181,120 @@ class Damage {
         RELEASE_ASSERT_NOT_REACHED();
     }
 
-    Region regionForTesting() const
-    {
-        Region region;
+    class iterator {
+        WTF_MAKE_FAST_ALLOCATED;
+    public:
+        iterator() = default;
 
-        // FIXME: we should not allow to create a Damage for an empty rect.
-        if (m_rect.isEmpty())
-            return region;
+        iterator(const Damage& damage, size_t index)
+            : m_damage(&damage)
+            , m_index(index)
+        {
+        }
 
+        iterator(const Damage& damage, BitVector::iterator iter)
+            : m_damage(&damage)
+            , m_iter(iter)
+            , m_index(*iter)
+        {
+        }
+
+        const IntRect& operator*()
+        {
+            switch (m_damage->m_mode) {
+            case Mode::Rectangles:
+                return m_damage->m_rects.rects.at(m_index);
+            case Mode::BoundingBox:
+                ASSERT(!m_index);
+                return m_damage->m_minimumBoundingRectangle;
+            case Mode::Full:
+                ASSERT(!m_index);
+                return m_damage->m_rect;
+            }
+            RELEASE_ASSERT_NOT_REACHED();
+        }
+
+        iterator& operator++()
+        {
+            if (m_iter) {
+                ++m_iter.value();
+                m_index = *m_iter.value();
+            } else
+                ++m_index;
+
+            return *this;
+        }
+
+        bool operator==(const iterator& other) const
+        {
+            return m_index == other.m_index;
```

Comment:
Shouldn't we also consider m_damage? Otherwise, I can create two iterators from two damage classes that could be equal.

Reply:
Fair enough